cmake/Reprojucer.cmake

```diff
@@ -2082,13 +2082,21 @@ function(__generate_icon_file icon_format out_icon_filename)
 endfunction()
 
 
-function(__set_output_directory_properties target)
+function(__set_output_directory_properties target subfolder)
 
   foreach(config ${JUCER_PROJECT_CONFIGURATIONS})
     string(TOUPPER "${config}" upper_config)
 
-    if(DEFINED JUCER_BINARY_LOCATION_${config})
+    if(MSVC AND NOT (DEFINED JUCER_VERSION AND JUCER_VERSION VERSION_LESS 5))
+      if(DEFINED JUCER_BINARY_LOCATION_${config})
+        set(output_directory "${JUCER_BINARY_LOCATION_${config}}/${subfolder}")
+      else()
+        set(output_directory "${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG>/${subfolder}")
+      endif()
+    elseif(DEFINED JUCER_BINARY_LOCATION_${config})
       set(output_directory "${JUCER_BINARY_LOCATION_${config}}")
+    endif()
+    if(output_directory)
       set_target_properties(${target} PROPERTIES
```

Comment:
Good catch! There is definitively a risk that `output_directory` is still defined, since this code is in a `foreach` loop. A previous iteration of the loop could have defined `output_directory` and then we would be using it wrongly. Fixed in ee2d689.